brave/brave-core
# Sync reset if browser was closed before the chain was fully created

https://github.com/brave/brave-browser/issues/6504

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/6504) for my issue if one did not already exist.
- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
From https://github.com/brave/brave-browser/issues/6504
1. Prepare clear profiles deviceA and deviceB
2. Launch profile A, create new sync chain, ensure sync code is shown, but don't copy the code.  Close deviceA.
3. Launch profile B, create new sync chain, copy sync code, don't close.
4. Launch profile A, enter sync code

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
